File: Exercises/Ex1/oppg4.py
from threading import Thread
# To compile and run
# python oppg4.py


i = 0

def threadFunc1():
	global i
	for j in range(0, 1000000):
		i+=1

def threadFunc2():
	global i
	for j in range(0, 1000000):
		i-=1

def main():
	#Initialize threads
	thread1 = Thread(target = threadFunc1, args = (), )
	thread2 = Thread(target = threadFunc2, args = (), )

	#Start threads
	thread1.start()
	thread2.start()

	#Join threads
	thread1.join()
	thread2.join()

	print(i)

main()

File: Exercises/Ex1/oppg4.c
#include <pthread.h>
#include <stdio.h>

/*To compile the file
gcc -pthread -o oppg4 oppg4.c
*/

int i = 0;

void *threadFunc1(){
	int j;
	for (j = 0; j < 1000000; ++j){
		++i;
	}
}

void *threadFunc2(){
	int j;
	for (j = 0; j < 1000000; ++j){
		--i;
	}
}

int main(void){
	//Initialize threads
	pthread_t thread1;
	pthread_t thread2;

	//Create threads
	pthread_create(&thread1, NULL, threadFunc1, "Creating thread 1"); //Create thread 1
	pthread_create(&thread2, NULL, threadFunc2, "Creating thread 2"); //Create thread 2

	//Join threads
	pthread_join(thread1, NULL);
	pthread_join(thread2, NULL);

	printf("%i\n", i);
	
	return 0;
}